iphone/Maps/Classes/CustomViews/NavigationDashboard/Views/RoutePreview/RoutePreviewStatus.swift

```diff
@@ -0,0 +1,130 @@
+
+  @IBOutlet private var errorBoxBottom: NSLayoutConstraint!
+  @IBOutlet private var resultsBoxBottom: NSLayoutConstraint!
+  @IBOutlet private var heightBoxBottom: NSLayoutConstraint!
+  @IBOutlet private var taxiBoxBottom: NSLayoutConstraint!
+
+  private var hiddenConstraint: NSLayoutConstraint!
+  weak var ownerView: UIView!
+
+  var isVisible = false {
+    didSet {
+      alternative(iPhone: {
+        guard self.isVisible != oldValue else { return }
+        if self.isVisible {
+          self.addView()
+        }
+        DispatchQueue.main.async {
+          guard let sv = self.superview else { return }
+          sv.setNeedsLayout()
+          self.hiddenConstraint.isActive = !self.isVisible
+          UIView.animate(withDuration: kDefaultAnimationDuration,
+                         animations: { sv.layoutIfNeeded() },
+                         completion: { _ in
+                          if (!self.isVisible) {
+                            self.removeFromSuperview()
+                          }
+          })
+        }
+      },
+                  iPad: { self.isHidden = !self.isVisible })()
+    }
+  }
+
+  private func addView() {
+    guard superview != ownerView else { return }
+    ownerView.addSubview(self)
+
+    NSLayoutConstraint(item: self, attribute: .left, relatedBy: .equal, toItem: ownerView, attribute: .left, multiplier: 1, constant: 0).isActive = true
+    NSLayoutConstraint(item: self, attribute: .right, relatedBy: .equal, toItem: ownerView, attribute: .right, multiplier: 1, constant: 0).isActive = true
+
+    hiddenConstraint = NSLayoutConstraint(item: self, attribute: .top, relatedBy: .equal, toItem: ownerView, attribute: .bottom, multiplier: 1, constant: 0)
+    hiddenConstraint.priority = UILayoutPriorityDefaultHigh
+    hiddenConstraint.isActive = true
+
+    let visibleConstraint = NSLayoutConstraint(item: self, attribute: .bottom, relatedBy: .equal, toItem: ownerView, attribute: .bottom, multiplier: 1, constant: 0)
+    visibleConstraint.priority = UILayoutPriorityDefaultLow
+    visibleConstraint.isActive = true
+  }
+
+  private func updateHeight() {
+    DispatchQueue.main.async {
+      self.setNeedsLayout()
+      self.errorBoxBottom.isActive = !self.errorBox.isHidden
+      self.resultsBoxBottom.isActive = !self.resultsBox.isHidden
+      self.heightBoxBottom.isActive = !self.heightBox.isHidden
+      self.taxiBoxBottom.isActive = !self.taxiBox.isHidden
+      UIView.animate(withDuration: kDefaultAnimationDuration) { self.layoutIfNeeded() }
+    }
+  }
+
+  func stateHidden() {
+    isVisible = false
+  }
+
+  func statePrepare() {
+    isVisible = false
+  }
+
+  func stateError(message: String) {
+    isVisible = true
+    errorBox.isHidden = false
+    resultsBox.isHidden = true
+    heightBox.isHidden = true
+    taxiBox.isHidden = true
+
+    errorLabel.text = message
+
+    updateHeight()
+  }
+
+  func stateReady() {
+    isVisible = true
```

Comment:
isHidden - default, responsible for view visibility.
isVisible - custom extension above isHidden, responsible for view appearance with animation.